Source/DafnyTestGeneration/TestMethod.cs

```diff
@@ -16,26 +18,34 @@ public class TestMethod {
     private static int nextId; // next unique id to be assigned
 
     // list of values to mock together with their types
-    public readonly List<(string id, Type type)> ObjectsToMock = new();
     // maps a variable that is mocked to its unique id
     private readonly Dictionary<DafnyModelVariable, string> mockedVarId = new();
     public readonly List<(string parentId, string fieldName, string childId)> Assignments = new();
+    public readonly List<(string id, Type type, string value)> ValueCreation = new();
+    private int nextValueId = 0;
     private readonly int id = nextId++;
     public readonly DafnyInfo DafnyInfo;
     // name of the method for which the counterexample is generated
     public readonly string MethodName;
     // values of the arguments to be passed to the method call
     public readonly List<string> ArgValues;
-    // number of type parameters for the method (all will be set to defaultType)
-    public readonly int NOfTypeParams;
+    // number of type arguments for the method (all will be set to defaultType)
+    public readonly int NOfTypeArgs;
     // default type to replace any type variable with
     private readonly Type defaultType = Type.Int;
     // the DafnyModel that describes the inputs to this test method
     private readonly DafnyModel dafnyModel;
-
     // Set of all types for which a {:synthesize} - annotated method is needed
     // These methods are used to get fresh instances of the corresponding types
-    private static readonly HashSet<string> TypesToSynthesize = new();
+    private static readonly Dictionary<string, UserDefinedType> TypesToSynthesize = new();
+    // is set to true whenever the tool encounters something it does not support
```

Comment:
This comment seems like it refers to an older version of this field.

Reply:
Thank you for catching this! I have updated the code to use a list instead and made sure the documentation is consistent (a set would make most sense but `UserDefinedType` is not hashable, so I was trying to get around that with a dictionary that maps a string representation to the object...)